driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedCrudHelper.java

```diff
@@ -1202,6 +1202,12 @@ private TimeSeriesOptions createTimeSeriesOptions(final BsonDocument timeSeriesD
                 case "metaField":
                     options.metaField(cur.getValue().asString().getValue());
                     break;
+                case "bucketMaxSpanSeconds":
+                    options.bucketMaxSpan(cur.getValue().asInt32().longValue(), TimeUnit.SECONDS);
+                    break;
+                case "bucketRoundingSeconds":
+                    options.bucketRounding(cur.getValue().asInt32().longValue(), TimeUnit.SECONDS);
```

Comment:
Why do we do `asInt32` here and above, given that the driver uses `new BsonInt64` in `CreateCollectionOperation.getCreateCollectionCommand`?

Reply:
Since we follow the JSON standard for our spec tests, it's not possible to explicitly indicate whether a numeric value should be interpreted as a `long` or an `int`. In our JSON parser, the value provided in the spec tests is treated as an `UNQUOTED_STRING`, and then the scanned digit is processed as follows:

```
long value = Long.parseLong(lexeme);
if (value < Integer.MIN_VALUE || value > Integer.MAX_VALUE) {
    return new JsonToken(JsonTokenType.INT64, value);
} else {
    return new JsonToken(JsonTokenType.INT32, (int) value);
}
```

You can refer to the code snippet in [JsonScanner.java#L449](https://github.com/vbabanin/mongo-java-driver/blob/d9aa6044e1a6b440bcb013c330497f2813484050/bson/src/main/org/bson/json/JsonScanner.java#L449) for more details.

The type determination is based on the actual value of the digit in the JSON file, which can be either an `int` or a `long`. In the case of the `bucketMaxSpan` and `bucketRounding` spec tests, `int `values are used so far.

If we were to always retrieve the value as an Int64, even when it is an Int32 in the JSON, we would encounter an error: org.bson.BsonInvalidOperationException: Value expected to be of type INT64 is of unexpected type INT32.

Similar cases can be found for maxTimeMs. You can refer to [UnifiedCrudHelper.java#L345](https://github.com/vbabanin/mongo-java-driver/blob/c633293f1dab303425969269bc93b9cc52bef96c/driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedCrudHelper.java#L345) and [FindOperation.java](https://github.com/vbabanin/mongo-java-driver/blob/0efc78c41c63e91da23532695fc31ee1dbd157d0/driver-core/src/main/com/mongodb/internal/operation/FindOperation.java#L414) for additional examples.

Reply:
Got it, thanks!